xtask/src/task_style.rs

```diff
@@ -0,0 +1,64 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::collections::BTreeSet;
+use std::io::{BufRead, BufReader};
+use std::process::{Command, Stdio};
+
+use anyhow::Result;
+
+use crate::util::*;
+
+fn check_test_names() -> Result<()> {
+    let wroot = workspace_root()?;
+
+    let mut cmd = Command::new("cargo");
+    let child = cmd
+        .args(["test", "--workspace", "--", "--list", "--format=terse"])
+        .stdin(Stdio::null())
+        .stdout(Stdio::piped())
+        .stderr(Stdio::null())
+        .current_dir(&wroot)
+        .spawn()?;
+
+    let problem_mods = BufReader::new(child.stdout.expect("stdout is present"))
+        .lines()
+        .map_while(std::result::Result::ok)
+        .filter_map(|line| {
+            // Look for "<test name>: test"
+            let test_name = match line.rsplit_once(": ") {
+                Some((p, "test")) => p,
+                _ => return None,
+            };
+
+            // skip doctests which follow "<path> - <testname> (line <num>)"
+            if test_name.contains(" - ") {
+                return None;
+            }
```

Comment:
nit, take it or leave it: we could, alternatively, filter out doctests by changing the command from `cargo test --workspace -- ...` to `cargo test --tests --lib --workspace -- ...`, to enable only lib tests and integration tests. that way, `cargo test` will never print the names of doctests (and, it might be a bit faster, as we also don't compile the doctests at all).

Reply:
Looks like `--all-targets` achieves this, so I'll rip out that complication in the parsing logic.